lib/shopping_cart.rb

```diff
@@ -0,0 +1,57 @@
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+  
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    {name: "#{name}", capacity: capacity_to_number}
```

Comment:
Since `name` is already a string, I don't think you need to interpolate this - you should be able to set the key/value with `name: name`